scopus-import/src/main/java/no/sikt/nva/scopus/conversion/files/ScopusFileConverter.java

```diff
@@ -239,21 +246,16 @@ private List<AssociatedArtifact> extractAssociatedArtifactsFromDoi(DocTp docTp)
     }
 
     private ScopusFile fetchFileContent(ScopusFile file) throws IOException {
-        var fetchFileResponse = fetchResponseAsInputStream(file.downloadFileUrl());
-        var filename = getFilename(fetchFileResponse);
-        var size = fetchFileResponse.body().readAllBytes().length;
+        var response = fetchResponseAsInputStream(file.downloadFileUrl());
+        var inputStream = tikaUtils.fetch(response.request().uri());
+        var filename = getFilename(response);
+        var size = inputStream.getLength();
         return file.copy()
-                   .withMimeType(getMimeType(fetchFileResponse.body()))
-                   .withContent(fetchFileResponse.body())
+                   .withContent(inputStream)
                    .withSize(size)
                    .withName(filename).build();
```

Comment:
Consuming inputstream two times. One for fetching response and extracting filename. Second one for fetching file using tika. `TikaInputStream + TikaConfig` seems to do much better work than `InputStream + Tikaconfig`.